molgenis-app/src/main/java/org/molgenis/app/promise/mapper/RadboudMapper.java

```diff
@@ -90,94 +57,36 @@ public MappingReport map(Entity project)
-					targetEntity.set(BIOBANK_SAMPLE_ACCESS_URI,
-							"http://www.radboudbiobank.nl/nl/collecties/materiaal-opvragen/");
-					targetEntity.set(WEBSITE, "http://www.radboudbiobank.nl/");
-					targetEntity.set(BIOBANK_DATA_ACCESS_URI,
-							"http://www.radboudbiobank.nl/nl/collecties/materiaal-opvragen/");
-					targetEntity.set(PRINCIPAL_INVESTIGATORS, getPrincipalInvestigator(biobankId));
-					targetEntity.set(INSTITUTES, getMrefEntities(REF_JURISTIC_PERSONS, "83"));
-
-					biobankExists = false;
+					bbmriSampleCollection = biobankMapper.mapNewBiobank(biobankEntity, samples, diseases);
+					dataService.add(SAMPLE_COLLECTIONS_ENTITY, bbmriSampleCollection);
+				}
+				else
+				{
+					bbmriSampleCollection = biobankMapper.mapExistingBiobank(biobankEntity, samples, diseases, bbmriSampleCollection);
+					dataService.update(SAMPLE_COLLECTIONS_ENTITY, bbmriSampleCollection);
 				}
-
-				targetEntity.set(ID, biobankId);
-				targetEntity.set(NAME, biobankEntity.getString(XML_TITLE));
-				targetEntity.set(TYPE, toTypes(biobankEntity.getString("TYPEBIOBANK")));
-				targetEntity.set(DATA_CATEGORIES, toDataCategories(biobankEntity, biobankSamplesEntities));
-				targetEntity.set(MATERIALS, toMaterials(biobankSamplesEntities));
-				targetEntity.set(OMICS, toOmics(biobankSamplesEntities));
-				targetEntity.set(SEX, toSex(biobankSamplesEntities));
-				targetEntity.set(AGE_LOW, toAgeMinOrMax(biobankSamplesEntities, true));
-				targetEntity.set(AGE_HIGH, toAgeMinOrMax(biobankSamplesEntities, false));
-				targetEntity.set(AGE_UNIT, getXrefEntity(REF_AGE_TYPES, "YEAR"));
-				targetEntity.set(DISEASE, getDiseaseTypes(biobankEntity.getString(XML_IDAA)));
-				targetEntity.set(NUMBER_OF_DONORS, size(biobankSamplesEntities));
-				targetEntity.set(DESCRIPTION, biobankEntity.getString(XML_DESCRIPTION));
-				targetEntity.set(CONTACT_PERSON, getContactPersons(biobankEntity));
-				targetEntity.set(BIOBANKS, getMrefEntities(REF_BIOBANKS, "RBB"));
-				targetEntity.set(BIOBANK_SAMPLE_ACCESS_FEE, true);
-				targetEntity.set(BIOBANK_DATA_ACCESS_JOINT_PROJECTS, true);
-				targetEntity.set(BIOBANK_DATA_SAMPLE_ACCESS_DESCRIPTION, null);  // Don't fill in
-				targetEntity.set(BIOBANK_DATA_ACCESS_FEE, true);
-				targetEntity.set(BIOBANK_DATA_ACCESS_JOINT_PROJECTS, true);
-				targetEntity.set(BIOBANK_DATA_ACCESS_DESCRIPTION, null);  // Don't fill in
-
-				if (biobankExists) sampleCollectionsToUpdate.add(targetEntity);
-				else sampleCollectionsToAdd.add(targetEntity);
 			});
 
-			LOG.info("Adding {} entities to {}", sampleCollectionsToAdd.size(), SAMPLE_COLLECTIONS_ENTITY);
-			dataService.add(SAMPLE_COLLECTIONS_ENTITY, sampleCollectionsToAdd.stream());
-
-			LOG.info("Updating {} entities in {}", sampleCollectionsToUpdate.size(), SAMPLE_COLLECTIONS_ENTITY);
-			dataService.update(SAMPLE_COLLECTIONS_ENTITY, sampleCollectionsToUpdate.stream());
-
 			LOG.info("Finished mapping RADBOUD biobanks");
 			report.setStatus(Status.SUCCESS);
```

Comment:
Probably would be informative to add a message as well, detailing how many samples and biobanks were processed?